File: Examples/test-suite/csharp/director_rev_len_str_runme.cs
using System;
using director_rev_len_strNamespace;

public class director_rev_len_str_runme
{
    static void Main()
    {
      Caller caller = new Caller(new DirectorReverseLengthStringCallback());
      int sum = caller.call();

      if (sum != 9*2*8)
        throw new Exception("Unexpected sum: " + sum);
    }
}

public class DirectorReverseLengthStringCallback : Callback {
    public DirectorReverseLengthStringCallback() : base() {}

    public override int run(string dataBufferAA)
    {
      int ret = 0;
      if (dataBufferAA != null) {
        for (int i = 0; i < dataBufferAA.Length; i++)
          ret += (int)dataBufferAA[i] * 2;
      }
      return ret;
    }
}

File: Examples/test-suite/perl5/director_rev_len_str_runme.pl
use strict;
use warnings;
use Test::More tests => 3;
BEGIN { use_ok 'director_rev_len_str' }
require_ok 'director_rev_len_str';

{
  package DirectorReverseLengthStringCallback;
  use base 'director_rev_len_str::Callback';
  sub run {
      my $ret = 0;
      my ($self, $dataBufferAA) = @_;
      if(defined $dataBufferAA) {
        $ret += ord($_) * 2 for(split('', $dataBufferAA));
      }
      return $ret;
  }
}

{
  print "Start\n";
  my $callback = DirectorReverseLengthStringCallback->new();
  my $caller = director_rev_len_str::Caller->new($callback);
  my $sum = $caller->call();

  is($sum, 9*2*8, 'Unexpected sum: ' . $sum);
}
